src/twisted/web/http.py

```diff
@@ -1035,6 +1039,10 @@ def write(self, data):
                             category=DeprecationWarning, stacklevel=2)
                         # Backward compatible cast for non-bytes values
                         value = networkString('%s' % (value,))
+                    if self.method == b"CONNECT" and name == b"Content-Length":
+                        log.msg("Warning: Removing Content-Length header in"
+                                " response to CONNECT request")
```

Comment:
`Content-Length` is totally unobjectionable in, say, a 4xx response to a CONNECT. It's only 2xx responses that are special.